### DISCUSSION TITLE
vue-sfc v3.3 syntax not supported ?

### DISCUSSION THREAD

**Discussion by takaoka-sr-co** [dotansimha/graphql-code-generator#9656] [Q&A]
codegen@5.0 use vue@3.3.4, but cannt use v3.3 defineProps() update.

1. open [vue/urql example](https://github.com/dotansimha/graphql-code-generator/tree/master/examples/vue/urql)

2. add import and defineProps() to `App.vue`
```ts
import { ExternalInterface } from '...';
defineProps<ExternalInterface>();
```
```ts
export interface ExternalInterface { opt?: string; }
```

3. run codegen

no error.
but missing `App.vue`'s query `AllFilmsWithVariablesQueryDocument`

----
other case...

1. add `*.graphql` to `codegen.ts`'s source

2. create `src/test/query1.graphql`, run codegen

new query added to gql.ts

3. create `src/test/Component1.vue` with query, run codegen

new query added to gql.ts

4. edit `Component1.vue` with defineProps<ExternalInterface>()

5. run codegen
    1. missing `Component1.vue`'s query (same as above)
    2. missing `query1.graphql`'s query (why)
    3. generate successfully all others

**Comment by Soren-Knudsen** [comment]
Same issue using the Apollo client
